utils/utils.go

```diff
@@ -350,6 +350,22 @@ func GetAbsPath(baseDir, path string) string {
 	return absPath
 }
 
+// ResolveHomeDir resolves prefix of the given path, if present, to the user's home directory and returns it.
+func ResolveHomeDir(filePath string) (string, error) {
+	if filePath == "" {
+		return "", nil
+	}
+	if homeDirPrefix := "~/"; strings.HasPrefix(filePath, homeDirPrefix) {
```

Comment:
Can `homeDirPrefix := "~/"` be a constant?

Reply:
also is there a way to check this only for linux and posix like systems? As in Windows this will not be the case.

Reply:
@mukundansundar on Windows also users can refer to %USERPROFILE%, do you want to not consider "~/" syntax? I propose to keep it the same as Linux and Mac.

<img width="999" alt="image" src="https://user-images.githubusercontent.com/22132944/222060366-cb52ea4d-60cd-45d7-a885-ae353f3857ad.png">